File: src/mffs/fortron/FortronHelper.java
package mffs.fortron;

import net.minecraftforge.fluids.FluidStack;
import net.minecraftforge.fluids.FluidTank;

/**
 * A class with useful functions related to Fortron.
 * 
 * @author Calclavia
 * 
 */
public class FortronHelper
{
	public static FluidStack FLUID_FORTRON;

	public static FluidStack getFortron(int amount)
	{
		FluidStack stack = FortronHelper.FLUID_FORTRON.copy();
		stack.amount = amount;
		return stack;
	}

	public static int getAmount(FluidStack liquidStack)
	{
		if (liquidStack != null)
		{
			return liquidStack.amount;
		}
		return 0;
	}

	public static int getAmount(FluidTank fortronTank)
	{
		if (fortronTank != null)
		{
			return getAmount(fortronTank.getFluid());
		}

		return 0;
	}
}

File: src/main/scala/mffs/FortronHelper.scala
package mffs

import net.minecraftforge.fluids.{Fluid, FluidStack, FluidTank}

/**
 * A class with useful functions related to Fortron.
 *
 * @author Calclavia
 */
object FortronHelper
{
  var FLUID_FORTRON: Fluid = null
  var FLUIDSTACK_FORTRON: FluidStack = null

  def getFortron(amount: Int): FluidStack =
  {
    val stack: FluidStack = new FluidStack(FLUID_FORTRON, amount)
    return stack
  }

  def getAmount(liquidStack: FluidStack): Int =
  {
    if (liquidStack != null)
    {
      return liquidStack.amount
    }
    return 0
  }

  def getAmount(fortronTank: FluidTank): Int =
  {
    if (fortronTank != null)
    {
      return getAmount(fortronTank.getFluid)
    }
    return 0
  }
}